client/state/selectors/get-active-discount.js

```diff
@@ -29,13 +29,22 @@ export const isDiscountActive = ( discount, state ) => {
 		return false;
 	}
 
-	if ( isPlainObject( discount.targetPlan ) ) {
+	if ( _.isPlainObject( discount.targetPlan ) ) {
```

Comment:
Maybe it's an unnecessary optimization, but you could have this block and the next one simply create an array (in this block with one element or in the next with multiple elements) and then run the `some()` check once over that array to avoid having to include it all twice.

If you include my comment about removing `map()` below, this whole two block section could then be something like the following:

```js
const targetPlans = isPlainObject( discount.targetPlan ) ? [ discount.targetPlan ] : ( Array.isArray( discount.targetPlans ) ? discount.targetPlans : [] );
const selectedSitePlanSlug = getSitePlanSlug( state, getSelectedSiteId( state ) );
return targetPlans.some( plan => planMatches( selectedSitePlanSlug, plan ) );
```

Bonus: you only have to import `isPlainObject` from lodash.